tfjs-backend-webgpu/src/webgpu_util.ts

```diff
@@ -56,6 +58,24 @@ export function computeDispatch(
                    (workGroupSize[2] * elementsPerThread[2])) :
                1
   ];
+
+  if (dispatchX <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE &&
+      dispatchY <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE &&
+      dispatchZ <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE) {
+    return [dispatchX, dispatchY, dispatchZ];
+  }
+
+  util.assert(dispatchX > MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE && !layout.y
+      && !layout.z, () => `Only support flatted layout to reshape dispatch ` +
+      `size when the dispatch size exceeds the limit size on x dimension.`);
+
+  let dispatchAverage = Math.ceil(Math.sqrt(dispatchX));
+  if (dispatchAverage > MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE) {
+    dispatchAverage = Math.ceil(Math.cbrt(dispatchX));
```

Comment:
Optionally:
```suggestion
    dispatchAverage = Math.ceil(Math.cbrt(dispatchX));
    util.assert(dispatchAverage <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE,
        () => 'Total dispatch size exceeds WebGPU maximum.');
```

(I'm sure this isn't really necessary because WebGPU will just throw a validation error anyway)